Library/ViewModels/CommentCellViewModel.swift

```diff
@@ -92,19 +92,29 @@ public final class CommentCellViewModel:
       .map(userIsBackingProject)
       .negate()
 
+    // If the user is either logged out, not backing or the flag is enabled, hide replyButton.
     self.replyButtonIsHidden = Signal.combineLatest(isLoggedOut, isNotABacker)
-      .map { isLoggedOut, isNotABacker in isLoggedOut || isNotABacker }
+      .map { isLoggedOut, isNotABacker in
+        (isLoggedOut || isNotABacker) || AppEnvironment.current.optimizelyClient?
+          .isFeatureEnabled(featureKey: OptimizelyFeature.Key.commentThreadingRepliesDisabled.rawValue) ??
+          false
+      }
 
-    // If both the replyButton and flagButton should be hidden, the entire stackview will be hidden too
+    // If both the replyButton and flagButton should be hidden, the entire stackview will be hidden too.
     self.bottomRowStackViewIsHidden = Signal.combineLatest(
       self.flagButtonIsHidden.signal,
       self.replyButtonIsHidden.signal
     ).map { flagButtonIsHidden, replyButtonIsHidden in
       flagButtonIsHidden && replyButtonIsHidden
     }
 
+    // If there are no replies or if the feature flag returns true, hide the stack view.
     self.viewRepliesStackViewIsHidden = comment.map(\.replyCount)
-      .map { $0 == 0 }
+      .map { replyCount in
```

Comment:
How about to improve readability here we break these out into private functions in this view model:

```swift
private func viewRepliesStackViewHidden(_ replyCount: Int) -> Bool {
  guard commentThreadingRepliesDisabled( == false else { return true }
  return replyCount == 0
}

private func commentThreadingRepliesDisabled() -> Bool {
  return AppEnvironment.current.optimizelyClient?.isFeatureEnabled(featureKey: OptimizelyFeature.Key.commentThreadingRepliesDisabled.rawValue) ?? false
}
```
And map over them like:

```swift
// If there are no replies or if the feature flag returns true, hide the stack view.
self.viewRepliesStackViewIsHidden = comment.map(\.replyCount)
  .map(viewRepliesStackViewHidden)
```

Reply:
Updated this and the replyButtonIsHidden logic as well. Both are quite similar.